var/spack/repos/builtin/packages/libaio/package.py

```diff
@@ -16,7 +16,7 @@ class Libaio(Package):
 
     def install(self, spec, prefix):
         # libaio is not supported on OS X
-        if spec.satisfies('arch=darwin-x86_64'):
+        if spec.platform == 'darwin' and spec.target.family == 'x86_64':
```

Comment:
I don't think darwin is used outside of x86_64, so the rest is redundant

Reply:
It's fine with me to remove the last condition. I'm definitely not a Mac expert, but I translated the constraint as it was given that up to ~10 years ago there was also support for `ppc`. Let me know what you think it should be done here.

Reply:
It looks like this package was added by @glennpj in #1049 if he wants to comment.

Based on the description: "This is the **linux** native ..." and the comment "not supported on **OS X**" I'm going to guess that the constraint is really about whether or not we are building on Linux. I would remove the `x86_64` part. Honestly, I wonder if we shouldn't just add a conflict for `platform=darwin` and then change the `depends_on` in other packages. Only `fio`, `lvm2`, and `mariadb` depend on `libaio`.

Reply:
Added conflict for `darwin` and modified `fio`, `lvm2` and `mariadb` in the process.